### PR TITLE
Consider `autoscaling.k8s.io/v1beta2` for CRD generation

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area control-plane
/kind regression

**What this PR does / why we need it**:
https://github.com/gardener/gardener/pull/8560 dropped static VPA CRDs but also completely removed the `autoscaling.k8s.io/v1beta2` version since it's not used/imported in `gardener/gardener`. We need to explicitly import this version to generate and deploy `VerticalPodAutoscaler` and `VerticalPodAutoscalerCheckpoint` of version `v1beta2`.

**Special notes for your reviewer**:
/invite @rfranzke 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Thanks!
/lgtm
/approve

File: hack/generate-crds.sh [hunk 1]
```diff
@@ -126,11 +118,11 @@ generate_group () {
     $generate
   fi
 
+  local files_in_use=("$@")
```
Comment (rfranzke) [inline hack/generate-crds.sh]:
```suggestion
  local desired_files=("$@")
```


or "relevant_files"

File: hack/generate-crds.sh [hunk 2]
```diff
@@ -126,11 +118,11 @@ generate_group () {
...
   while IFS= read -r crd; do
     crd_out="$output_dir/$file_name_prefix$(basename $crd)"
-    if [ "$crd" != "$crd_out" ]; then
-      mv "$crd" "$crd_out"
-    fi
+    mv "$crd" "$crd_out"
+    files_in_use+=("$(basename "$crd_out")")
```
Comment (rfranzke) [inline hack/generate-crds.sh]:
```suggestion
    desired_files+=("$(basename "$crd_out")")
```

File: hack/generate-crds.sh [hunk 3]
```diff
@@ -149,7 +141,28 @@ generate_group () {
...
       sed -i "/^  annotations:.*/a\    api-approved.kubernetes.io: $k8s_io_api_approval_reason" "$crd_out"
     fi
-  done < <(ls "$output_dir/${group/hvpa/}"_*.yaml)
+  done < <(ls "$output_dir_temp/${group/hvpa/}"_*.yaml)
+
+  # garbage collection - clean all generated files for this group to account for changed prefix or removed resources
+  local pattern=".*${group}_.*\.yaml"
+  if [[ "$group" == "autoscaling.k8s.io" ]]; then
+    pattern=".*${group}_v.*\.yaml"
+  fi
```
Comment (rfranzke) [inline hack/generate-crds.sh:150]: Why is this needed? Isn't the only difference the `v` in the `pattern`? The "general" pattern in L147 should already cover files that match `.*${group}_v`?

File: hack/generate-crds.sh [hunk 4]
```diff
@@ -149,7 +141,28 @@ generate_group () {
...
+      file_name=$(basename "$crd")
+      if [[ $file_name == "$file_name_in_use" ]] || [[ ! $file_name =~ $pattern ]]; then
+        delete=false
+        break
+      fi
+    done
+
+    if $delete; then
+      rm "$crd"
+    fi
```
Comment (rfranzke) [inline hack/generate-crds.sh]:
```suggestion
  while IFS= read -r file; do
    file_name="$(basename "$file")"
    delete_no_longer_needed_file=true

    for desired_file in "${desired_files[@]}"; do
      if [[ $file_name == "$desired_file" ]] || [[ ! $file_name =~ $pattern ]]; then
        delete_no_longer_needed_file=false
        break
      fi
    done

    if $delete_no_longer_needed_file; then
      rm "$file"
    fi
```

File: hack/generate-crds.sh (hunk 3 above)
Comment (timuthy) [reply hack/generate-crds.sh:150]: If I see it correctly, you contributed this https://github.com/gardener/gardener/pull/8560/files#diff-1fb280337f572205ceb830c22a7bf50c6b26dbed29c8425f198083f5bf6b95e6R101-R109. So you tell me? 😄

Comment (rfranzke) [reply hack/generate-crds.sh:150]: 😂 Ah, it is because HVPA wrongly uses the same group like VPA. OK, then let's just keep it... 😵

**Review by rfranzke — COMMENTED** [review]
/lgtm
